api/app/controllers/api/v1/positions_controller.rb

```diff
@@ -1,92 +1,83 @@
 # frozen_string_literal: true
 
 module Api::V1
-  # Controller for Positions
-  class PositionsController < ApplicationController
-    before_action :set_position, only: %i[show update destroy]
+    # Controller for Positions
+    class PositionsController < ApplicationController
 
-    # GET /positions
-    def index
-      @positions = Position.order(:id)
-
-      render json: @positions
-    end
-
-    # GET /positions/1
-    def show
-      render json: @position
-    end
-
-    # POST /positions
-    def create
-      @position = Position.new(position_params)
-
-      if @position.save
-        render json: @position, status: :created
-      else
-        render json: @position.errors, status: :unprocessable_entity
-      end
-    end
-
-     # POST /positions/import
-     def import
-      @position = Position.new(position_params)
-
-      if @position.valid?
-        if @position.save
-          render json: @position, status: :created
-        else
-          render json: @position.errors, status: :unprocessable_entity
-        end
-      else
-        #check whether the error lies in uniqueness alone
-        if @position.errors.details.length == 1 &&  @position.errors["course_code"] == ["course duplicated in same session"]
-          # safe to update since there's no other errors
-          @exists = Position.where('course_code = ? AND session_id = ?', params[:course_code], params[:session_id])
-          if @exists.update(position_params)
-            render json: @exists
-          else
-            render json: @exists.errors, status: :unprocessable_entity
-          end
-          
-        else
-          render json: @position.errors, status: :unprocessable_entity
+        # GET /positions
+        def index
+            render json: { status: 'success', message: '', payload: positions_by_session }
         end
-      end
-    end
 
-    # PATCH/PUT /positions/1
-    def update
-      if @position.update(position_params)
-        render json: @position
-      else
-        render json: @position.errors, status: :unprocessable_entity
-      end
-    end
+        # POST /positions
+        def create
+            req_keys = [:position_code, :position_title]
+            if Session.exists?(id: params[:session_id])
```

Comment:
I think it makes code more readable to have early returns instead of nested `if`s.

e.g.,
```
if not Session.exists?(id: params[:session_id])
    render json: { status: 'error', message: 'Invalid session_id', payload: {} }
    return
end
if not (req_keys & params.keys.map(&:to_sym) == req_keys)
    render json: { status: 'error',  message: 'Missing position_code or position_title', payload: {} }
    return
end

position = Position.new(position_params)
...
```

Reply:
I've removed the nested if statements.